cmake/modules/AddCephTest.cmake

```diff
@@ -1,5 +1,22 @@
 #AddCephTest is a module for adding tests to the "make check" target which runs CTest
 
+execute_process(
+    COMMAND uname -m
+    OUTPUT_VARIABLE ARCHSTR
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+if(LINUX)
+  set(OSSTR linux)
+elseif(FREEBSD)
+  execute_process(
+    COMMAND uname -r
+    OUTPUT_VARIABLE OSVER
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+  set(OSSTR freebsd-${OSVER})
+endif(LINUX)
```

Comment:
i think that's exactly the expected behaviour on unsupported platforms. and the the error message will guide the user to the right place (here) to fix it.

if we continue with an empty `$OSSTR`, the error is hidden at this moment, and user will be lost when deciphering the error messages when the tests fail.